pkgs/node/src/chain.zig

```diff
@@ -461,7 +467,114 @@ pub const BeamChain = struct {
+                std.fmt.fmtSliceHexLower(&vote.head.root),
+            });
+            return AttestationValidationError.UnknownHeadBlock;
+        };
+
+        const source_block = self.forkChoice.protoArray.nodes.items[source_idx];
+        const target_block = self.forkChoice.protoArray.nodes.items[target_idx];
+        const head_block = self.forkChoice.protoArray.nodes.items[head_idx];
+        _ = head_block; // Will be used in future validations
+
+        // 2. Validate slot relationships
+        if (source_block.slot > target_block.slot) {
+            self.module_logger.debug("Attestation validation failed: source slot {d} > target slot {d}", .{
+                source_block.slot,
+                target_block.slot,
+            });
+            return AttestationValidationError.SourceSlotExceedsTarget;
+        }
+
+        //    This corresponds to leanSpec's: assert vote.source.slot <= vote.target.slot
+        if (vote.source.slot > vote.target.slot) {
+            self.module_logger.debug("Attestation validation failed: source checkpoint slot {d} > target checkpoint slot {d}", .{
+                vote.source.slot,
+                vote.target.slot,
+            });
+            return AttestationValidationError.SourceCheckpointExceedsTarget;
+        }
+
+        // 3. Validate checkpoint slots match block slots
+        if (source_block.slot != vote.source.slot) {
+            self.module_logger.debug("Attestation validation failed: source block slot {d} != source checkpoint slot {d}", .{
+                source_block.slot,
+                vote.source.slot,
+            });
+            return AttestationValidationError.SourceCheckpointSlotMismatch;
+        }
+
+        //    This corresponds to leanSpec's: assert target_block.slot == vote.target.slot
+        if (target_block.slot != vote.target.slot) {
+            self.module_logger.debug("Attestation validation failed: target block slot {d} != target checkpoint slot {d}", .{
+                target_block.slot,
+                vote.target.slot,
+            });
+            return AttestationValidationError.TargetCheckpointSlotMismatch;
+        }
+
+        // 4. Validate attestation is not too far in the future
+        //    Per leanSpec store.py:
+        //    - General validation (line 140): assert vote.slot <= Slot(current_slot + Slot(1))
+        //    - Gossip processing (line 177): assert vote.slot <= time_slots
+        //
+        //    Gossip attestations must be for current or past slots only. Validators vote
+        //    in interval 1 of the current slot, so they cannot vote for future slots.
+        //    Block attestations can be more lenient since the block itself was validated.
+        const current_slot = self.forkChoice.fcStore.timeSlots;
+        const max_allowed_slot = if (is_from_block)
+            current_slot + constants.MAX_FUTURE_SLOT_TOLERANCE // Block attestations: allow +1
```

Comment:
```suggestion
            current_slot - constants.MAX_FUTURE_SLOT_TOLERANCE // Block attestations: allow +1
```

this is the correct condition, i.e. an attestation/vote packed in a proposal block at a slot can't be any later than the previous slot

can you also do the PR in the spec to fix this if thats there

Reply:
In the spec where have this  `assert vote.slot <= Slot(current_slot + Slot(1)), "Attestation too far in future"` which means it should be `assert vote.slot <= Slot(current_slot - Slot(1)), "Attestation from current or future slot"`  Line https://github.com/leanEthereum/leanSpec/blob/161c0493e980c4f26b70c3f873d0f17b378f6268/src/lean_spec/subspecs/forkchoice/store.py#L178  gossip check would become redundant after this fix (since -1 is stricter than <= current_slot), so should I leave it  as a redundant safety check?